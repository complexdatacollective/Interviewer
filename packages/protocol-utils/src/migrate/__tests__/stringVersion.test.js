/* eslint-env jest */

const migrateProtocol = require('../migrateProtocol');

const v1Protocol = {
  schemaVersion: '1.0.0',
  codebook: {
    node: {
      disallowedType: {
        name: 'disallowed type name',
        variables: {
          invalidExampleVariable: {
            name: 'variable (with) disallowed characters!',
          },
          invalidExampleVariableDuplicate: {
            // should resolve to the same as previous, therefore needs incrementing
            name: 'variable (with) disallowed characters?',
          },
        },
      },
    },
  },
};

describe('string version handling', () => {
  it('migrates version "1.0.0" correctly', () => {
    const [result] = migrateProtocol(v1Protocol, 4);

    expect(result).toMatchSnapshot();
  });

  it('throws an error for other source string version numbers', () => {
    expect(() => {
      migrateProtocol({
        ...v1Protocol,
        schemaVersion: '2.0.0',
      }, 4);
    }).toThrow('The source schema version is not recognised, must be an integer ("2.0.0").');
  });

  it('throws an error for target string version numbers', () => {
    expect(() => {
      migrateProtocol(v1Protocol, '4.0.0');
    }).toThrow('The target schema version is not recognised, must be an integer ("4.0.0").');
  });
});
