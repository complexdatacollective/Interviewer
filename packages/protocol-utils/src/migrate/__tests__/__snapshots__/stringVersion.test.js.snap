// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string version handling migrates version "1.0.0" correctly 1`] = `
Object {
  "codebook": Object {
    "node": Object {
      "disallowedType": Object {
        "name": "disallowed_type_name",
        "variables": Object {
          "invalidExampleVariable": Object {
            "name": "variable_with_disallowed_characters",
          },
          "invalidExampleVariableDuplicate": Object {
            "name": "variable_with_disallowed_characters2",
          },
        },
      },
    },
  },
  "schemaVersion": 4,
  "stages": Array [],
}
`;
